packages/neuron-ui/src/components/Settings/Wallets.tsx

```diff
@@ -52,55 +54,42 @@ const UpdateWallet = styled(WalletButton)`
   margin: 30px 0 0 50px;
 `
 const General = () => {
-  const wallets: string[] = [
-    'Wallet Name1',
-    'Wallet Name2',
-    'Wallet Name3',
-    'Wallet Name4',
-    'Wallet Name5',
-    'Wallet Name6',
-  ]
-  const [walletSelected, setWalletSelected] = useState(wallets[0])
-  const [walletsState, setWalletsState] = useState(wallets)
+  const walletNames = useContext(WalletNamesContext)
+  const [walletSelected, setWalletSelected] = useState(0)
+
+  useEffect(() => {
+    UILayer.getWalletNameListStore()
+  }, walletNames.name)
 
   const deleteWallet = () => {
-    const temps = walletsState
-    const temp = walletSelected
-    const index = temps.indexOf(temp)
-    if (index > -1) {
-      temps.splice(index, 1)
-    }
-    setWalletSelected(temps[0])
-    setWalletsState(temps)
+    UILayer.deleteWalletStore(walletNames.name[walletSelected])
+    setWalletSelected(0)
   }
 
   const addWallet = () => {
-    const temps = walletsState
-    setWalletsState(temps.concat(`Wallet Name${temps.length + 1}`))
+    const walletName = `wallet${Math.ceil(Math.random() * 100)}`
+    UILayer.saveWalletStore(walletName, {
+      name: walletName,
+      keystore: '',
+    })
   }
 
   const updateWallet = () => {
-    const temps = walletsState
-    const temp = walletSelected
-    const index = temps.indexOf(walletSelected)
-    const name = prompt('Please enter your wallet name', temp)
-    if (index > -1 && name != null && name !== '') {
-      temps.splice(index, 1, name)
-    }
-    setWalletsState(temps)
+    const walletName = `wallet${Math.ceil(Math.random() * 100)}`
+    UILayer.renameWalletStore(walletName, walletNames.name[walletSelected])
   }
 
   return (
     <ContentPanel>
-      {walletsState.map(wallet => {
-        if (walletSelected === wallet) {
+      {walletNames.name.map(wallet => {
```

Comment:
`Settings/Wallets` should keep a copy of wallet objects, not names. We'll need to access the wallet object for more information.